src/environment.cpp

```diff
@@ -1544,6 +1544,49 @@ void ServerEnvironment::step(float dtime)
 		*/
 		removeRemovedObjects();
 	}
+
+	/*
+		Manage particle spawner expiration
+	*/
+	if (m_particle_management_interval.step(dtime, 1.0)) {
+		for (std::map<u32, float>::iterator i = m_particle_spawners.begin();
+				i != m_particle_spawners.end(); ) {
+			//non expiring spawners
+			if (i->second <= -1.f) {
+				++i;
+				continue;
+			}
+
+			i->second -= 1.0f;
+			if (i->second <= 0.f)
+				m_particle_spawners.erase(i++);
+			else
+				++i;
+		}
+	}
+}
+
+u32 ServerEnvironment::addParticleSpawner(float exptime)
+{
+	// Timers with lifetime 0 do not expire
+	float time = exptime > 0.f ? exptime : -1.f;
```

Comment:
In reference to the above note/comment
> <est31> Zeno`, i dont think so
> <est31> err
> <est31> you are right
> <est31> sorry you are right :)

And discussions earlier in the day I've amended the code to use PARTICLE_SPAWNER_NO_EXPIRY and modified lines 1555 and 1572 (in this obsolete version of the commit... line numbers in latest are 1157 and 1574)